File: programmers/Level 1/소수 찾기/solution.java
class Solution {
  public int solution(int n) {
      int answer = 0;
      int dp[] = new int [1000001];
      
      for(int i=2; i*i <= n; ++i){
          if(dp[i]==0)
          {
              for(int j=i+i; j<=n; j+=i)
                  dp[j] = 1;
          }
      }
      for(int i=2; i<=n; ++i){
          if(dp[i]==0){
              answer++;
          }
      }
      return answer;
  }
}

File: programmers/Level 1/소수 찾기/solution.js
function solution(n) {
    var answer = 0;
    var dp = new Array();
    
    for(var i=2; i<=n; ++i)
        dp[i] = 0;

    for(var i=2; i*i<=n; ++i){
        if(dp[i]==0){
            for(var j=i+i; j<=n; j+=i)
                dp[j]=1;
        }
    }

    for(var i=2; i<=n; ++i){
        if(dp[i]==0)
            answer++;
    }
    return answer;
}